### DISCUSSION TITLE
How to add field for hidden attribute?

### DISCUSSION THREAD

**Discussion by diazsasak** [filamentphp/filament#1472] [Help] [answered]
### Package

filament/forms

### Package Version

v2.9.16

### Laravel Version

v8.70.2

### Livewire Version

v2.8.0

### PHP Version

php 8.0.14

### Bug description

I have hidden attribute because it need to be hidden on API response (json).

![image](https://user-images.githubusercontent.com/7880137/152669570-5ec6d913-f996-4fd3-b257-eec7142f2ce1.png)
![image](https://user-images.githubusercontent.com/7880137/152669576-37e36c0f-e437-414b-8f1a-97b38fec06a4.png)
![image](https://user-images.githubusercontent.com/7880137/152669582-5f65592e-1171-4c05-a0f0-7229ad2f5938.png)

if I remove from hidden and use ->makeVisible on my API the pagination is gone. Any solution?

Thank you for this productive package

**Comment by danharrin** [comment, marked as answer]
On your resource's edit page, have you tried this inside the `beforeFill()` method?

```php
protected function beforeFill(): void
{
    $this->record->makeVisible(['minimal_stock'])
}
```

**Reply by diazsasak** [reply]
It's worked. 

Thank you very much, i need to read more carefully on the docs
